### PR TITLE
Define all infrastructure using @guardian/cdk

### PR DESCRIPTION
**Pull request by jacobwinch**
## What does this change?

This replaces all CFN-defined infrastructure with new infrastructure defined via `@guardian/cdk`, based on [the following guide](https://github.com/guardian/cdk/blob/jw-scheduled-lambda-migration-docs/docs/migration-guide-scheduled-lambda.md).

## How to test

Unfortunately there is no `CODE` environment for this project, so I have deployed this to `PROD` and confirmed that all logs/metrics are still coming through as expected.

## How can we measure success?

We have completed another migration to `@guardian/cdk` and have successfully rehearsed the documentation added in https://github.com/guardian/cdk/pull/1179.

## Have we considered potential risks?

Yes, as mentioned above the lack of `CODE` environment is not ideal for risk mitigation.

In fact, I broke this Lambda for a brief period of time when deploying to `PROD` (for testing purposes) as the [code change](https://github.com/guardian/elastic-search-monitor/blob/7fa8cd7dc3b4185a2d1bb160a237884d0e68db2b/src/main/scala/com/gu/elasticsearchmonitor/Lambda.scala) is not shipped until after the CFN update completes.

Unfortunately AWS seems to take ~30 minutes to delete a Lambda that is provisioned inside a VPC, so I updated the function code manually to minimise the downtime here. A good reminder that `CODE` environments are very useful ways to discover these types of hard to predict problems!

### CODE REVIEW

**Review by akash1810 — APPROVED** [review]

File: cdk/lib/elastic-search-monitor.ts [hunk 1]
```diff
@@ -1,15 +1,74 @@
...
+      fileName: "elastic-search-monitor.jar",
+      environment: {
+        TAG_QUERY_APP: "elk-es-master",
+        TAG_QUERY_STACK: "deploy",
+        CLUSTER_NAME: "elk",
+      },
+      description:
+        "Monitors your Elasticsearch cluster and reports metrics to CloudWatch",
+      handler: "com.gu.elasticsearchmonitor.Lambda::handler",
+      monitoringConfiguration: { noMonitoring: true },
```
Comment (nicl) [inline cdk/lib/elastic-search-monitor.ts:56]: Is it worth turning monitoring on (perhaps in a later PR?) or is there some reason it is not useful here?

**Review by nicl — APPROVED** [review]
🚢

File: cdk/lib/elastic-search-monitor.ts (hunk 1 above)
Comment (jacobwinch) [reply cdk/lib/elastic-search-monitor.ts:56]: We should probably turn it on in case we ever break the lambda - will open a follow-up PR for that.

Comment (jacobwinch) [reply cdk/lib/elastic-search-monitor.ts:56]: I should have also mentioned that we currently define some alarms [in a separate template](https://github.com/guardian/elastic-search-monitor/blob/e0d3703bb5819f12ac9f398c97c3a738eb3f3a68/alarms.yaml). Based on the [`riff-raff.yaml`](https://github.com/guardian/elastic-search-monitor/blob/e0d3703bb5819f12ac9f398c97c3a738eb3f3a68/riff-raff.yaml) I think these must be deployed manually, so I will try to migrate those across too.

Comment (jacobwinch) [reply cdk/lib/elastic-search-monitor.ts:56]: https://github.com/guardian/elastic-search-monitor/pull/18

**Pull request merged**